h/search/query.py

```diff
@@ -251,56 +223,53 @@ class DeletedFilter(object):
     deleted.
     """
 
-    def __call__(self, _):
-        return {"bool": {"must_not": {"exists": {"field": "deleted"}}}}
+    def __call__(self, search, _):
+        return search.exclude("exists", field="deleted")
 
 
 class NipsaFilter(object):
     def __init__(self, request):
         self.group_service = request.find_service(name='group')
         self.user = request.user
 
-    def __call__(self, _):
-        return nipsa_filter(self.group_service, self.user)
-
+    def __call__(self, search, _):
+        return self._nipsa_filter(search, self.group_service, self.user)
 
-def nipsa_filter(group_service, user=None):
-    """Return an Elasticsearch filter for filtering out NIPSA'd annotations.
+    def _nipsa_filter(self, search, group_service, user=None):
+        """Return an Elasticsearch filter for filtering out NIPSA'd annotations.
 
-    The returned filter is suitable for inserting into an Es query dict.
-    For example::
+        The returned filter is suitable for inserting into an Es query dict.
```

Comment:
Any reason not to fold `_nipsa_filter` into the `__call__` method above? Also the documentation is out of date (it doesn't return a filter dict any more). I suggest just removing the no-longer-relevant parts.